utils.bal

```diff
@@ -0,0 +1,669 @@
+    json payload = check fileResource.cloneWithType(json);
+    string path = prepareUrlWithCopyOptional(fileId, optional);
+    json response = check sendRequestWithPayload(httpClient, path, payload);
+    File|error file = response.cloneWithType(File);
+    if (file is File) {
+        return file;
+    } else {
+        return error(ERR_FILE_RESPONSE, file);
+    }
+}
+
+# Update file metadata using the fileID.
+# 
+# + httpClient - The HTTP Client
+# + fileId - ID of the file to be updated
+# + optional - 'UpdateFileMetadataOptional' used to add query parameters to the request
+# + fileResource - 'File' can added as a payload to change metadata
+# + return - If successful, returns `File`. Else returns `error`
+function updateFileById(http:Client httpClient, string fileId, UpdateFileMetadataOptional? optional = (), 
+                        File? fileResource = ()) returns @tainted File|error {
+    json payload = check fileResource.cloneWithType(json);
+    string path = prepareUrlWithUpdateOptional(fileId, optional);
+    json response = check updateRequestWithPayload(httpClient, path, payload);
+    File|error file = response.cloneWithType(File);
+    if (file is File) {
+        return file;
+    } else {
+        return error(ERR_FILE_RESPONSE, file);
+    }
+}
+
+# Create new file (with only metadata).
+# 
+# + httpClient - The HTTP Client
+# + optional - 'CreateFileOptional' used to add query parameters to the request
+# + fileData - 'File' Metadata is send to in the payload 
+# + return - If successful, returns `File`. Else returns `error`
+function createMetaDataFile(http:Client httpClient, CreateFileOptional? optional = (), File? fileData = ()) 
+                                returns @tainted File|error {
+    json payload = check fileData.cloneWithType(json);
+    string path = prepareUrlwithMetadataFileOptional(optional);
+    json response = check uploadRequestWithPayload(httpClient, path, payload);
+    File|error file = response.cloneWithType(File);
+    if (file is File) {
+        return file;
+    } else {
+        return error(ERR_FILE_RESPONSE, file);
+    }
+}
+
+# Upload new file.
+# 
+# + httpClient - The HTTP Client
+# + filePath - Path to the file object to be uploaded
+# + optional - 'UpdateFileMetadataOptional' used to add query parameters to the request
+# + fileMetadata - 'File' Metadata is send to in the payload 
+# + return - If successful, returns `File`. Else returns `error`
+function uploadFile(http:Client httpClient, string filePath, UpdateFileMetadataOptional? optional = (), 
+                                File? fileMetadata = ()) returns @tainted File|error {    
+    string path = prepareUrl([UPLOAD, DRIVE_PATH, FILES]);  
+    json response = check uploadFiles(httpClient, path, filePath);  
+    //update metadata
+    json|error responseId = response.id;
+    string fileId = EMPTY_STRING;
+    if (responseId is json) {
+        fileId = responseId.toString();
+    }
```

Comment:
Can't we use this?
```
json responseId = check response.id;
fileId = responseId.toString();
  
```

Reply:
json responseId = check response.id;
    string fileId = responseId.toString();
    
    
    Changed as above.
    Thanks Chuhaashanan